viz/visualize.py

```diff
@@ -370,39 +266,46 @@ def all_latent_traversals(self, sample_latent_space=None, size=8, filename='imgs
         size : int
             Number of samples for each latent traversal.
         """
-        #TODO Need to sort the visualisation issue here, we no longer use KL divergence in the same way for all losses
+        # TODO Need to sort the visualisation issue here, we no longer use KL divergence in the same way for all losses
         # https://github.com/YannDubs/disentangling-vae/pull/25#issuecomment-473535863
         
         latent_samples = []
 
-        avg_kl_list = read_avg_kl_from_file(os.path.join(self.model_dir, 'losses.log'), self.model.latent_dim)
+        avg_kl_list = read_avg_kl_from_file(os.path.join(self.model_dir, 'losses.log'))
 
         # Perform line traversal of every latent
         for idx in range(self.model.latent_dim):
             latent_samples.append(self.latent_traverser.traverse_line(idx=idx,
                                                                       size=size,
                                                                       sample_latent_space=sample_latent_space))
-        latent_samples = [
-            latent_sample for _, latent_sample in sorted(zip(avg_kl_list, latent_samples), reverse=True)
-        ]
-        sorted_avg_kl_list = [
-            round(float(avg_kl_sample), 3) for avg_kl_sample, _ in sorted(zip(avg_kl_list, latent_samples), reverse=True)
-        ]
-
-        # Decode samples
-        generated = self._decode_latents(torch.cat(latent_samples, dim=0))
 
         if self.save_images:
-            traversal_images_with_text = add_labels(
-                label_name='KL',
-                tensor=generated,
-                num_rows=size,
-                sorted_list=sorted_avg_kl_list,
-                dataset=self.dataset)
-
-            traversal_images_with_text.save(filename)
-            return avg_kl_list
+            if self.reorder_latent_dims:
+                latent_samples = [
+                    latent_sample for _, latent_sample in sorted(zip(avg_kl_list, latent_samples), reverse=True)
+                ]
+                sorted_avg_kl_list = [
+                    round(float(avg_kl_sample), 3) for avg_kl_sample, _ in sorted(zip(avg_kl_list, latent_samples), reverse=True)
+                ]
+
+                # Only decode the top 5 KL dimensions
+                generated = self._decode_latents(torch.cat(latent_samples[:5], dim=0))
+
+                traversal_images_with_text = add_labels(
```

Comment:
woudln't it be so much nicer to just:
1. return always (don't use `if self.save_images`)
2. sort if needed
3. save 
(again as said before that wiuld be nicer but it's kind of everywhere so don't feel obliged to do it if it takes too much time)